lightning_client.go

```diff
@@ -3469,3 +3475,32 @@ func (s *lightningClient) QueryRoutes(ctx context.Context,
 		TotalAmtMsat:  lnwire.MilliSatoshi(route.TotalAmtMsat),
 	}, nil
 }
+
+func (s *lightningClient) CheckMacaroonPermissions(ctx context.Context,
+	macaroon []byte, permissions []MacaroonPermission, fullMethod string) (bool,
+	error) {
+
+	rpcPermissions := make([]*lnrpc.MacaroonPermission, len(permissions))
```

Comment:
You're right, the second argument is only optional for maps. You have another problem here now though: this will create a slice with `len(permissions)` nil elements. And then in the loop you `append` more elements, effectively doubling the size of the slice. It's a weird behavior of Golang...

So either you have this pattern (setting the `size` to 0 but the `capacity` to the target length):

```
	rpcPermissions := make([]*lnrpc.MacaroonPermission, 0, len(permissions))
	for _, perm := range permissions {
		permission := &lnrpc.MacaroonPermission{
			Entity: perm.Entity,
			Action: perm.Action,
		}

		rpcPermissions = append(rpcPermissions, permission)
	}
```

or this one (using the index directly):

```
	rpcPermissions := make([]*lnrpc.MacaroonPermission, len(permissions))
	for idx, perm := range permissions {
		rpcPermissions[idx] = &lnrpc.MacaroonPermission{
			Entity: perm.Entity,
			Action: perm.Action,
		}
	}
```

I think both patterns are equally efficient but the second one is often preferred if you know the length beforehand.

Reply:
got it, thanks for the explanation!